bgpd/bgp_attr.c

```diff
@@ -2890,7 +2890,7 @@ bgp_attr_parse_ret_t bgp_attr_parse(struct peer *peer, struct attr *attr,
 			 * a stack buffer, since they perform bounds checking
 			 * and we are working with untrusted data.
 			 */
-			unsigned char ndata[BGP_MAX_PACKET_SIZE];
+			unsigned char ndata[peer->max_packet_size];
```

Comment:
Having a local variable with 65K may have issues when stack size is set low. Should we make it static or allocate the memory?

Reply:
how do you suggest?

Reply:
making it static would not keep it in stack alternatively we can have scratch pads which can be reused.

Reply:
I think I mentioned this already and someone reminded me that the default stack size is 2mb.

Obviously this might still be an issue for embedded systems or something but I think we're okay.

Reply:
> I think I mentioned this already and someone reminded me that the default stack size is 2mb.
> 
> Obviously this might still be an issue for embedded systems or something but I think we're okay.

Agree VLAs can't be static but we could use some pre-allocated chunks for the same? Or do you mean if someone wants to run it on embedded systems with lesser stack size then do not use this feature which is anyway configurable.